dali/pipeline/data/types.h

```diff
@@ -156,7 +157,9 @@ class DLL_PUBLIC TypeInfo {
    *        (only relevant for device and host pinned memory)
    */
   template <typename DstBackend, typename SrcBackend>
-  DLL_PUBLIC void Copy(void *dst, const void *src, Index n, cudaStream_t stream,
```

Comment:
Please keep the doc in sync with the arguments.

Can you elaborate in the docstring about the intended behaviour? I see one comment:
```
// assume same device if either is not set
```
Is it a fallback or an intended option? Should we not provide device_ids only when we are on CPU or it's ok to skip them if it's the same device?